### ISSUE TITLE
OTA via dashboard and arduino IDE

### ISSUE THREAD

**Issue by Michiel-FMI-ImProvia** [espressif/esp-rainmaker#161]
Is it possible to do the OTA updates via the https://dashboard.insights.espressif.com when using the Arduino IDE? 
up until now I get an Error when using RMaker.enableOTA(OTA_USING_TOPICS); and the ESP reboots

**Comment by JohnMacrae** [comment]
I raised it in 2021 here( https://github.com/espressif/arduino-esp32/issues/5420) It has been studiously ignored despite prodding

**Comment by Michiel-FMI-ImProvia** [comment]
Yeah, I saw your item. It would be amazing if it was possible to upload via the dashboard and all the devices get the new firmware. Another option is to try it via the ESP-IDF but then I need time to learn how this works. It's just that there is some OTA code in the arduino examples, and it bugs me that it seems to have no functionality.

**Comment by shahpiyushv** [comment]
Low heap memory seems to be the root cause for this. Since Arduino has pre compiled libraries which are common across all esp32 specific modules, there are quite some things pulled in even when not used actively. Getting to a stage wherein the RAM usage will be reduced, but without affecting any other functionality/examples will take some time.

**Comment by JohnMacrae** [comment]
It works fine in idf.

**Comment by Michiel-FMI-ImProvia** [comment]
> Low heap memory seems to be the root cause for this. Since Arduino has pre compiled libraries which are common across all esp32 specific modules, there are quite some things pulled in even when not used actively. Getting to a stage wherein the RAM usage will be reduced, but without affecting any other functionality/examples will take some time.

I can imagine that this is not a simple problem. Maybe mention this in the Arduino examples to not get people's hopes up.
In short  if I want to use Rainmaker OTA it's smarter to learn the ESP-IDF language.

**Comment by JohnMacrae** [comment]
There is a workaround @Michiel-FMI-ImProvia . Use Arduino OTA https://youtu.be/V-g45CFSFr4

**Comment by Michiel-FMI-ImProvia** [comment]
Oh I am so exited!! Will watch that video with intense attention!!!

**Comment by JohnMacrae** [comment]
I hope this will give some impetus to the Espressif team. If the Arduino
OTA works the surely it's not a heap collision?

**Comment by me-chiel** [comment]
exactly....

**Comment by sanketwadekar** [comment]
Hi @JohnMacrae,
We are aware of this crash and are working on the fixes. We will fix this issue in the upcoming release of ESP32 Arduino.
Regards,
Sanket

**Comment by Michiel-FMI-ImProvia** [comment]
> Hi @JohnMacrae, We are aware of this crash and are working on the fixes. We will fix this issue in the upcoming release of ESP32 Arduino. Regards, Sanket

any idea when this release will happen? This functionality sounds so amazing.

**Comment by sanketwadekar** [comment]
Hello,
Arduino 2.0.6 has been released, which fixes the OTA crash issues seen previously. If you face low memory errors like 
`esp-tls-mbedtls: mbedtls_ssl_setup returned -0x7F00`, please reboot your device once. Doing a reboot after successful provisioning and then starting OTA is recommended.
Thanks!
